deploy/lvm-operator.yaml

```diff
@@ -123,14 +123,28 @@ spec:
           description: VolStatus string that specifies the current state of the volume
             provisioning request.
           properties:
+            error:
```

Comment:
do we need this now? we can directly rely on state being failed to take the decesion?

Reply:
Yeah, functionally from rescheduling point of view it can be removed now. I had just kept it for the purpose of better inspection as user/maintainer. As we are propagating error message to csi external provisioner (along with ResourceExhausted), the same error message is being added as events under pvc resource. e.g 
```
➜  lvm-localpv git:(storage-capacity) ✗ k describe pvc claim-a-test-a-4
Name:          claim-a-test-a-4
Namespace:     default
StorageClass:  openebs-ext4pv
Status:        Pending
Volume:
Labels:        app=test-a
Annotations:   volume.beta.kubernetes.io/storage-provisioner: local.csi.openebs.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:
Access Modes:
VolumeMode:    Filesystem
Used By:       test-a-4
Events:
  Type     Reason                Age               From                                                                                Message
  ----     ------                ----              ----                                                                                -------
  Normal   WaitForFirstConsumer  22s               persistentvolume-controller                                                         waiting for first consumer to be created before binding
  Normal   Provisioning          5s (x4 over 22s)  local.csi.openebs.io_openebs-lvm-controller-0_6f7c487b-6397-4d19-a5d4-b4d84819f831  External provisioner is provisioning volume for claim "default/claim-a-test-a-4"
  Normal   ExternalProvisioning  5s (x5 over 22s)  persistentvolume-controller                                                         waiting for a volume to be created, either by external provisioner "local.csi.openebs.io" or manually created by system administrator
  Warning  ProvisioningFailed    3s (x4 over 19s)  local.csi.openebs.io_openebs-lvm-controller-0_6f7c487b-6397-4d19-a5d4-b4d84819f831  failed to provision volume with StorageClass "openebs-ext4pv": rpc error: code = ResourceExhausted desc =   Volume group "lvmvg" has insufficient free space (3071 extents): 5120 required.
 - exit status 5
  Normal  WaitForPodScheduled  3s (x5 over 19s)  persistentvolume-controller  waiting for pod test-a-4 to be scheduled
```